### PR TITLE
Add shopping cart test

### PR DESCRIPTION
**Pull request by RomanSoproniuk**
controller
repository
service

### CODE REVIEW

File: src/test/java/book/store/onlinebookstore/repository/BookRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    public void findAllByCategoriesId_ReturnAllBookByCategoryId_ShouldReturnOneBook() {
+        //given
+        Long categoryId = 1L;
+        Long expectedBookId = 1L;
+        String expectedBookTitle = "Kobzar";
+        //when
+        List<Book> allByCategoriesId = bookRepository.findAllByCategoriesId(pageable, categoryId);
+        Long actualId = allByCategoriesId.get(0).getId();
+        String actualTitle = allByCategoriesId.get(0).getTitle();
```
Comment (MAMentorFX) [inline src/test/java/book/store/onlinebookstore/repository/BookRepositoryTest.java:38]: extract allByCategoriesId.get(0) to local variable.

File: src/test/java/book/store/onlinebookstore/controller/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                .setDescription("Ukr");
+        //when
+        MvcResult result = mockMvc.perform(get("/categories/{id}", categoryId))
+                .andExpect(status().isOk())
+                .andReturn();
+        CategoryDto actual = objectMapper
+                .readValue(result.getResponse().getContentAsString(), CategoryDto.class);
+        //then
+        Assertions.assertNotNull(actual.getId());
+        Assertions.assertEquals(actual.getId(), categoryId);
```
Comment (MAMentorFX) [inline src/test/java/book/store/onlinebookstore/controller/CategoryControllerTest.java:144]: extract to local variable actual.getId()

File: src/test/java/book/store/onlinebookstore/service/BookServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        expectedDtoWithOutCategory.setId(1L);
+        expectedDtoWithOutCategory.setTitle("Kobzar");
+        expectedDtoWithOutCategory.setAuthor("Shechenko");
+        expectedDtoWithOutCategory.setPrice(BigDecimal.valueOf(50L));
+        expectedSecondDtoWithOutCategory = new BookDtoWithoutCategoryIds();
+        expectedSecondDtoWithOutCategory.setId(2L);
+        expectedSecondDtoWithOutCategory.setTitle("Lisova Mavka");
+        expectedSecondDtoWithOutCategory.setAuthor("Ukrainka");
+        expectedSecondDtoWithOutCategory.setPrice(BigDecimal.valueOf(100L));
+    }
```
Comment (MAMentorFX) [inline src/test/java/book/store/onlinebookstore/service/BookServiceTest.java:95]: unreadable, use constructors. Also suggest @BeforeAll for such logic. Do the same for all similar test classes

File: src/test/resources/database/books/add-book-for-delete.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO books (id, title, author, price, isbn, description, cover_image)
+VALUES (3, 'SadokKoloHatu', 'Shevchenko', 50, '43623uefhef723', 'Awesome', 'SuperImage');
```
Comment (MAMentorFX) [inline src/test/resources/database/books/add-book-for-delete.sql:2]:
```suggestion
VALUES (3, 'SadokKoloHatu', 'Shevchenko', 50, '43623uefhef723', 'Awesome', 'SuperImage');

```
dont forget empty line. Do the same for each .sql

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

File: src/test/java/book/store/onlinebookstore/service/BookServiceTest.java (hunk 3 above)
Comment (RomanSoproniuk) [reply src/test/java/book/store/onlinebookstore/service/BookServiceTest.java:95]: Screenshot_1
Is that better, for real?

Comment (okuzan) [reply src/test/java/book/store/onlinebookstore/service/BookServiceTest.java]: this is more or less a matter of personal code style preference, but here I tend to the second option, setters are just too much visual clutter. You can opt to use builders if you don't like constructors, seems like a compromise

**Review by okuzan — CHANGES_REQUESTED** [review]
why the change?
image

**Review by okuzan — APPROVED** [review]

**Pull request merged**